### PR TITLE
Git ignore phpstorm stuff, composer binary

### PR DESCRIPTION
**Pull request by gggeek**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,6 @@ build/
 logs/
 nbproject/
 vendor/
+.idea/
```
Comment (aik099) [inline .gitignore]: I have this in my global `.gitignore`.

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,6 @@ build/
...
 composer.lock
+composer.phar
```
Comment (aik099) [inline .gitignore]: In fact I recommend installing composer globally to avoid this problem altogether.

File: .gitignore [hunk 3]
```diff
@@ -2,4 +2,6 @@ build/
 logs/
 nbproject/
 vendor/
+.idea/
 composer.lock
```
Comment (aik099) [inline .gitignore]: We'd better add (as a separate PR) the `composer.lock` to repo to ensure that we're using same dependency versions for development and on Travis.

Comment (gggeek) [inline .gitignore]: I agree about composer.lock.
Can remove it if we agree

File: .gitignore (hunk 2 above)
Comment (gggeek) [inline .gitignore]: still, having it here does not harm anyone - it allows both usages of composer (installed globally or locally).
same for phpstrom files ignore...

File: .gitignore (hunk 1 above)
Comment (gggeek) [inline .gitignore]: Well, I work with many OS and VMs, often having the same php app mounted under both linux and windows.
Thus I do not have always the same global environment configuration available

Comment (aik099) [inline .gitignore]: A valid point. As you said it doesn't harm to have it.

File: .gitignore (hunk 2 above)
Comment (aik099) [inline .gitignore]: A valid point. As you said it doesn't harm to have it.

**Comment by aik099** [comment]
@robocoder , any objections preventing this PR to be merged?

File: .gitignore (hunk 1 above)
Comment (uuf6429) [inline .gitignore]: You can also put it in `.git/info/exclude` instead - which is a local per-repository setting.

**Pull request merged**